### ISSUE TITLE
Coding Style Guide Discussion

### ISSUE THREAD

**Issue by dodekeract** [dotModded/ContinuumShader#4]
As [the wiki-entry](https://github.com/dotModded/ContinuumShader/wiki/Coding-Style-Guidelines) states, that the style guidelines can be discussed I'm opening this issue/thread.

While I know that spaces vs tabs is a quite opinionated topic I'm a little bit confused, why @DethRaid used [the greatest pro-tab-argument as a pro-spaces argument](https://github.com/dotModded/ContinuumShader/wiki/Coding-Style-Guidelines#4-use-spaces-instead-of-tab-characters). As nearly everybody has a personal preference on how far the code has to be intended the use of the tab-character is exactly the solution to that. Everybody can set his editor to the preferred tab-width, while not messing with other people's settings. The main point of that character nowadays actually is, that every editor can decide on how wide it should be. This pleases programmers who prefer 4-wide and programmers who like 2- or even 8-wide indentations. Another advantage is that, no matter if your editor supports "making 4 spaces appear as one character when navigating through code" or not, with tabs you can always easily navigate and change indentation.

While IMHO tabs are far superior compared to spaces, (because of being customizable) I do get that a personal opinionated choice is involved in choosing between the two. Therefore *my main critique is, that a pro-tabs argument is the only reason this project might use spaces*, which is quite weird.

TLDR: Wiki uses pro-tabs argument against tabs.

**Comment by DethRaid** [comment]
My argument is that spaces are always rendered consistently, so the person who sees your code sees it the same way as you did when you wrote it. You consider the fact that tabs can be rendered differently to be a good thing, while I think it's a bad thing.

**Comment by dodekeract** [comment]
Well - nothing prevent a user from setting it to 4 wide. If your editor doesn't support that, then it most likely doesn't support that "easy-navigate-through-spaces, 4 at a time" either. And in case of indentation - why is it important that it looks the same for everybody instead of everybody just defining his preferred indentation width?
